File: 203-remove-linked-list-elements/remove-linked-list-elements.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeElements(ListNode head, int val) {
        
        ListNode prevNode = null;
        var currentNode = head;
        
        while (currentNode != null) {
            
            if (currentNode.val != val) {
                prevNode = currentNode;
                currentNode = currentNode.next;
                continue;
            }
                
            // Remove the node
            if (prevNode == null) {
                
                head = currentNode.next;
                final var nextNode = currentNode.next;
                currentNode.next = null;
                currentNode = nextNode;
                continue;
            }
            
            prevNode.next = currentNode.next;
            currentNode.next = null;
            currentNode = prevNode.next;
        }
        
        return head;
    }
}

File: 203-remove-linked-list-elements/remove-linked-list-elements.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeElements(self, head: Optional[ListNode], val: int) -> Optional[ListNode]:
        if not head:
            return head
        
        prev_node = None
        current_node = head
        
        while current_node:

            if current_node.val != val:
                prev_node = current_node
                current_node = current_node.next
                continue
            

            # Remove the element
            if not prev_node:
                head = current_node.next
                current_node = head
                continue
            
            prev_node.next = current_node.next
            current_node = prev_node.next
        
        return head
